test/test_scenario_perception_pipeline.cpp

```diff
@@ -7,92 +7,165 @@
 #include <moveit/moveit_cpp/planning_component.h>
 
 #include <geometry_msgs/msg/pose.hpp>
+#include <nav_msgs/msg/path.hpp>
+#include <dynmsg/msg_parser.hpp>
+#include <dynmsg/typesupport.hpp>
+#include <dynmsg/yaml_utils.hpp>
+
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <iostream>
 
 using moveit::planning_interface::MoveGroupInterface;
 
-// class MoveGroupServer {
-// };
+const std::string PLANNING_GROUP = "panda_arm";
 
-class ScenarioPerceptionPipeline {
+class ScenarioPerceptionPipeline
+{
 public:
-    ScenarioPerceptionPipeline(std::shared_ptr<MoveGroupInterface>);
-    bool go_to_poses(std::vector<geometry_msgs::msg::Pose> pose_list);
+  ScenarioPerceptionPipeline(std::shared_ptr<MoveGroupInterface>);
+  std::tuple<int, int> runTestCase(const nav_msgs::msg::Path& pose_list);
+  bool sendTargetPose(const geometry_msgs::msg::Pose& target_pose);
+
 private:
-    std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
+  std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
 };
 
-ScenarioPerceptionPipeline::ScenarioPerceptionPipeline(std::shared_ptr<MoveGroupInterface> move_group_interface_ptr) 
-: move_group_interface_ptr_(move_group_interface_ptr) { }
+ScenarioPerceptionPipeline::ScenarioPerceptionPipeline(std::shared_ptr<MoveGroupInterface> move_group_interface_ptr)
+  : move_group_interface_ptr_(move_group_interface_ptr)
+{
+}
 
-bool ScenarioPerceptionPipeline::go_to_poses(std::vector<geometry_msgs::msg::Pose> pose_list) {
-    for (auto& pose : pose_list) {
-        move_group_interface_ptr_->setPoseTarget(pose);
+std::tuple<int, int> ScenarioPerceptionPipeline::runTestCase(const nav_msgs::msg::Path& test_case)
+{
+  int success_number = 0;
+  int failure_number = 0;
+  for (auto& pose_stamped : test_case.poses)
+  {
+    bool is_successful = sendTargetPose(pose_stamped.pose);
+    if (is_successful)
+      success_number++;
+    else
+      failure_number++;
+  }
+
+  return { success_number, failure_number };
+}
 
-        moveit::planning_interface::MoveGroupInterface::Plan plan_msg;
-        auto const ok = static_cast<bool>(move_group_interface_ptr_->plan(plan_msg));
+bool ScenarioPerceptionPipeline::sendTargetPose(const geometry_msgs::msg::Pose& target_pose)
+{
+  move_group_interface_ptr_->setPoseTarget(target_pose);
 
-        if (ok) {
-            move_group_interface_ptr_->execute(plan_msg);
-        } else {
-            return false;
-        }
-    }
+  moveit::planning_interface::MoveGroupInterface::Plan plan_msg;
+  const auto ok = static_cast<bool>(move_group_interface_ptr_->plan(plan_msg));
 
+  if (ok)
+  {
+    move_group_interface_ptr_->execute(plan_msg);
```

Comment:
Do we need the execution at all? If not you could just return `move_group_interface_ptr_->plan(plan_msg)` with from this function

Reply:
I couldn't understand your question. Wasn't our goal to test all pipeline rather than just planning pipeline? By the way i think the effectively measuring of middleware benchmarking is proportional to data-transfering works. I'm wondering your ideas about it.